File: leetcode/tree/sum_of_left_leaves.py
def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
    return self.dfs(root, False)

def dfs(self, node, left_side):
    if not node:
        return 0
    elif not node.left and not node.right:
        if left_side:
            return node.val
        else:
            return 0
    else:
        return self.dfs(node.left, True) + self.dfs(node.right, False)

File: leetcode/tree/sum_of_left_leaves.cpp
int sumOfLeftLeaves(TreeNode* root) {
	return f(root, false);
}
int f(TreeNode* root, bool is_true) {
	if (!root)
		return 0;
	else if (!root->right && !root->left) {
		if (is_true) 
			return root->val;
		else
			return 0;
	}
	else
		return f(root->right, false) + f(root->left, true);
}
